node/node.go

```diff
@@ -371,23 +454,29 @@ func (nn nexNode) Ping() (*actorproto.PingNodeResponse, error) {
 		return nil, err
 	}
 	for _, c := range agentSuper.Children() {
-		agentResp, err := agentSuper.Ask(nn.ctx, c, &actorproto.QueryWorkloads{})
+		agentResp, err := agentSuper.Ask(nn.ctx, c, &actorproto.QueryWorkloads{}, 5*time.Second)
 		if err != nil {
 			nn.options.Logger.Error("Failed to ping agent", slog.Any("err", err))
 			return nil, errors.New("failed to ping agent")
 		}
-		aR, ok := agentResp.(*actorproto.WorkloadList)
+		aREnv, ok := agentResp.(*actorproto.Envelope)
 		if !ok {
-			nn.options.Logger.Error("Failed to convert agent response")
-			return nil, errors.New("failed to convert agent response")
+			nn.options.Logger.Error("Failed to convert envelope")
+			return nil, errors.New("failed to convert envelope")
+		}
+		var aR actorproto.WorkloadList
+		err = aREnv.Payload.UnmarshalTo(&aR)
+		if err != nil {
+			nn.options.Logger.Error("Failed to unmarshal agent response")
+			return nil, errors.New("failed to unmarshal agent response")
 		}
 		resp.RunningAgents[c.Name()] = int32(len(aR.Workloads))
 	}
 
 	return resp, nil
 }
 
-func (nn nexNode) GetInfo() (*actorproto.NodeInfo, error) {
+func (nn nexNode) GetInfo(namespace string) (*actorproto.NodeInfo, error) {
```

Comment:
All the methods on `nexNode` should be pointer receivers, e.g. `nn *nexNode` to avoid unnecessary copying of the struct fields.